[AArch64] Add FEAT_FPAC to Neoverse V2

sjoerdmeijer opened the pull request:
This feature is supported in Grace, but wasn't specified in the CPU definition. This is not important for codegen, but is good for completeness, and good for other tools that could query the CPU definition (e.g. llvm-exegesis).

Jojad9 reviewed (approved):
Approve to this change

rj-jesus commented on llvm/lib/Target/AArch64/AArch64Processors.td:
```diff
@@ -1067,7 +1067,8 @@ def ProcessorFeatures {
                                      FeatureJS, FeatureLSE, FeatureRAS, FeatureRCPC, FeatureRDM];
-  list<SubtargetFeature> Grace = !listconcat(NeoverseV2, [FeatureSVE2SM4, FeatureSVEAES, FeatureSVE2SHA3]);
+  list<SubtargetFeature> Grace = !listconcat(NeoverseV2, [FeatureSVE2SM4, FeatureSVEAES, FeatureSVE2SHA3,
+                                                          FeatureFPAC]);
```
Hi, does it make sense to move this to the Neoverse V2 definition?

sjoerdmeijer replied:
Thanks for spotting this, I think you're right.  I got confused with "supported", thinking about this architecturally, but this is in the V2 TRM, so all V2 cores implement this. I will move this to the V2.

davemgreen replied:
Yeah that sounds good - there are a number of other cores that have it but are not marked, but I don't have a list. The features was added after the cores were added and it didn't add them to all. Adding for V2 is a good start.

It can change the way PACBTI is emitted, I believe to make it more efficient and shouldn't be an issue so long as the instructions do throw exceptions.

rj-jesus commented on llvm/lib/Target/AArch64/AArch64Processors.td:
```diff
@@ -555,7 +555,8 @@ def TuneNeoverseV2 : SubtargetFeature<"neoversev2", "ARMProcFamily", "NeoverseV2
                                       FeatureAvoidLDAPUR,
-                                      FeaturePredictableSelectIsExpensive]>;
+                                      FeaturePredictableSelectIsExpensive,
+                                      FeatureFPAC]>;
```
Is there any reason you placed this in tuning instead of the main processor features of the Neoverse V2?

sjoerdmeijer replied:
Nope. :-( 
But now fixed.

sjoerdmeijer committed:
[AArch64] Add FEAT_FPAC to Neoverse V2

This feature is supported in Grace, but wasn't specified in the CPU
definition.

rj-jesus reviewed (approved):
Except for a seemingly out-of-order test, LGTM!

sjoerdmeijer commented on clang/test/Driver/print-enabled-extensions/aarch64-grace.c:
```diff
@@ -19,6 +19,7 @@
 // CHECK-NEXT:     FEAT_ETE                                               Enable Embedded Trace Extension
 // CHECK-NEXT:     FEAT_FCMA                                              Enable Armv8.3-A Floating-point complex number support
 // CHECK-NEXT:     FEAT_FHM                                               Enable FP16 FML instructions
+// CHECK-NEXT:     FEAT_FPAC                                              Enable Armv8.3-A Pointer Authentication Faulting enhancement
```
Thanks, yeah, I have been messing around with that file a lot and tried different things just to get rid of this Git warning:

> -// CHECK-NEXT:     FEAT_VHE                                               Enable Armv8.1-A Virtual Host extension
> \ No newline at end of file
> +// CHECK-NEXT:     FEAT_VHE                                               Enable Armv8.1-A Virtual Host extension

I don't get it, I don't see what is wrong.
But it was already there, I am ignoring it for now.

rj-jesus replied:
I think the file was missing a new line character at the end of the file (which presumably your editor added automatically when you added the `FEAT_FPAC` line). I think this is a harmless change, but if you want to undo it, I believe `truncate -s -1` should work.
